src/streamlit_extras/exception_handler/__init__.py

```diff
@@ -0,0 +1,100 @@
+"""
+Override Streamlit's global uncaught exception handler.
+
+This extra exposes a single function, ``set_global_exception_handler``, that replaces
+Streamlit's uncaught exception handler with a user-provided callable. It supports
+Streamlit versions before and after 1.39.0, where the handler location changed.
+"""
+
+from __future__ import annotations
+
+import sys
+from typing import Callable
+
+import streamlit as st
+from packaging.version import Version
+
+from .. import extra
+
+__all__ = ["set_global_exception_handler"]
+
+
+@extra
+def set_global_exception_handler(f: Callable) -> None:
+    """Replace Streamlit's global uncaught exception handler with ``f``.
+
+    Args:
+        f (Callable): New exception handler function.
+
+    Notes:
+        - For Streamlit versions earlier than 1.39.0, the handler lives under
+          ``streamlit.runtime.scriptrunner.script_runner``.
+        - For Streamlit 1.39.0 and later, it resides under ``streamlit.error_util``.
+
+    Warning:
+        This function mutates Streamlit internals. Use carefully and test thoroughly.
+    """
+
+    if Version(st.__version__) < Version("1.39.0"):
+        parent_module = sys.modules["streamlit.runtime.scriptrunner.script_runner"]
+    else:
+        parent_module = sys.modules["streamlit.error_util"]
+
+    parent_module.handle_uncaught_app_exception.__code__ = f.__code__
```

Comment:
nit: Is the replacement of the __code__ instead of just changing the handle_uncaught_app_exception reference - e.g. `parent_module.handle_uncaught_app_exception = f` - required for this to work?

Reply:
I don't really know tbh, this has been working in the past - mostly borrowed code code from Joshua